src/librustc_resolve/late.rs

```diff
@@ -2038,35 +2038,19 @@ impl<'a, 'b> LateResolutionVisitor<'a, '_> {
         }
 
         for &(trait_name, binding) in traits.as_ref().unwrap().iter() {
-            // Traits have pseudo-modules that can be used to search for the given ident.
-            if let Some(module) = binding.module() {
-                let mut ident = ident;
-                if ident.span.glob_adjust(
-                    module.expansion,
-                    binding.span,
```

Comment:
This is not the right span to determine whether the trait is in scope, it's a span of the whole import or the whole item, while we need to use a span of a single identifier, like `span(Foo)` in `use Trait as Foo;`.
Otherwise we can't correctly process imports constructed from parts, e.g.
```rust
#![feature(decl_macro)]

mod m {
    pub trait Tr {
        fn method(&self) {}
    }

    impl Tr for u8 {}
}

macro gen_import($Br: ident) {
    use m::Tr as $Br;
}
gen_import!(Br);

fn main() {
    0u8.method(); // ERROR: no method named `method`, despite passing `Br` from the outside
}
```